bodhi/tests/functional/test_updates.py

```diff
@@ -1955,6 +1955,33 @@ def test_revoke_action_for_testing_request(self, publish, *args):
         publish.assert_called_with(
                 topic='update.request.revoke', msg=mock.ANY)
 
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.notifications.publish')
+    def test_obsolete_update_on_pending_state(self, publish, *args):
```

Comment:
I'm not familiar with the rest of the tests in this area, so can you please ensure that these five conditions are covered by tests that are either new or existing? There are 4 conditions that will be tested in the new if statement, so we could conceivably test 5 conditions. The first is a positive case (basically, what you have here but modified to set the autokarma and reach the negative threshold), and the other four just ensure the negative cases:

1. The update is pending, is marked for testing, has autokarma true, and has reached the negative karma threshold. This update should get obsoleted.
2. All of the first test, but the update isn't pending.
3. All of the first test, but the update isn't marked for testing (this one should probably also be obsoleted, but by another function in the code. I also bet there's already a test for this.)
4. All of the above, but autokarma is off.
5. All of the above, but the karma is not beyond the thresholds.

You don't necessarily have to write the above tests if they already exist, but please ensure that they do exist.